src/main/java/uk/gov/hmcts/juror/api/moj/report/standard/PoolAnalysisReport.java

```diff
@@ -0,0 +1,145 @@
+package uk.gov.hmcts.juror.api.moj.report.standard;
+
+import com.querydsl.core.Tuple;
+import com.querydsl.jpa.impl.JPAQuery;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.juror.api.moj.controller.reports.request.StandardReportRequest;
+import uk.gov.hmcts.juror.api.moj.controller.reports.response.AbstractReportResponse;
+import uk.gov.hmcts.juror.api.moj.controller.reports.response.StandardReportResponse;
+import uk.gov.hmcts.juror.api.moj.controller.reports.response.StandardTableData;
+import uk.gov.hmcts.juror.api.moj.domain.QJurorPool;
+import uk.gov.hmcts.juror.api.moj.domain.QPoolRequest;
+import uk.gov.hmcts.juror.api.moj.report.AbstractStandardReport;
+import uk.gov.hmcts.juror.api.moj.report.DataType;
+import uk.gov.hmcts.juror.api.moj.repository.CourtLocationRepository;
+import uk.gov.hmcts.juror.api.moj.utils.NumberUtils;
+import uk.gov.hmcts.juror.api.moj.utils.SecurityUtil;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Component
+public class PoolAnalysisReport extends AbstractStandardReport {
+
+    private final CourtLocationRepository courtLocationRepository;
+
+    @Autowired
+    public PoolAnalysisReport(CourtLocationRepository courtLocationRepository) {
+        super(
+            QJurorPool.jurorPool,
+            DataType.POOL_NUMBER_BY_JP,
+            DataType.SERVICE_START_DATE,
+            DataType.JURORS_SUMMONED_TOTAL,
+            DataType.RESPONDED_TOTAL,
+            DataType.ATTENDED_TOTAL,
+            DataType.PANEL_TOTAL,
+            DataType.JUROR_TOTAL,
+            DataType.EXCUSED_TOTAL,
+            DataType.DISQUALIFIED_TOTAL,
+            DataType.DEFERRED_TOTAL,
+            DataType.REASSIGNED_TOTAL,
+            DataType.UNDELIVERABLE_TOTAL,
+            DataType.TRANSFERRED_TOTAL,
+            DataType.FAILED_TO_ATTEND_TOTAL
+        );
+
+        this.courtLocationRepository = courtLocationRepository;
+    }
+
+    @Override
+    public Class<? extends Validators.AbstractRequestValidator> getRequestValidatorClass() {
+        return RequestValidator.class;
+    }
+
+    @Override
+    protected void preProcessQuery(JPAQuery<Tuple> query, StandardReportRequest request) {
+        query.where(QPoolRequest.poolRequest.returnDate.between(request.getFromDate(), request.getToDate()));
```

Comment:
As we are grouping by Juror pool this would be better as

QJurorPool.jurorPool.pool.returnDate

(Same for all instances where you grab pool)

Reply:
updated